apps/rush-lib/src/api/LastInstallFlag.ts

```diff
@@ -4,6 +4,141 @@ import { FileSystem, JsonFile, JsonObject } from '@microsoft/node-core-library';
 
 export const LAST_INSTALL_FLAG_FILE_NAME: string = 'last-install.flag';
 
+/** @internal */
+interface IPropertyErrors {
+  /** The properties to check and force an abort on inequality */
+  [key: string]: {
+    /**
+     * List of additional properties between both states that must match
+     * in order to abort with an error
+     */
+    deps?: string[];
+    /** The error to throw if conditions are met */
+    error: string;
+  };
+}
+
+/** @internal */
+interface ILastInstallFlagErrorState {
+  oldState: JsonObject;
+  newState: JsonObject;
+}
+
+/**
+ * Provides error checking utilities, and defines an `Error` of type `LastInstallFlagError`
+ * that can be generated by `LastInstallFlag` when it is desired.
+ * 
+ * If `LastInstallFlag.isValid` is called with `abortOnInvalid = true`, then the properties
+ * in `LastInstallFlagError._errors` will be used to find which properties between the old
+ * and new `LastInstallFlag` states are not equal, and then generate a friendly error for 
+ * that particular property.
+ * @internal
+ */
+export class LastInstallFlagError extends Error {
+  /**
+   * Collection of properties that can generate an error if an inequality
+   * is detected between old & new state.
+   * If `deps` are defined for a property, all of those properties must match
+   * between old & new state in order for the error message to be generated.
+   */
+  private static _errors: IPropertyErrors = {
+    storePath: {
+      error: "Current PNPM store path does not match the last one used.  This may cause inconsistency in your builds.\n\n" +
+      "If you wish to install with the new store path, please run \"rush update --purge\"",
+      deps: ["packageManager"]
+    }
+  }
```

Comment:
I think it would be simpler if we just stuck these inline in the `switch` statement in the constructor.

Reply:
I simplified the whole process for generating errors for the `last-install.flag` state.  Basically ripped out all of the abstraction, so it's much more straightforward now.